### DISCUSSION TITLE
Type inference is not possible in html templates.

### DISCUSSION THREAD

**Discussion by Shin-Beomchul** [vuejs/pinia#1225] [Help and Questions] [answered]
### Reproduction

https://github.com/Shin-Beomchul/mono-vue3-module-federation/blob/master/applications/dml/src/views/examples/PiniaStoreExample.vue

### Steps to reproduce the bug

I am working on a transition from vue2 to vue3. However, it is the same source code, but only in the vue3 <template></template> area.
 Pinia type inference is not possible.

Type inference is not possible in html templates.
The difference between these two is the difference in vue version.
Does vue3 not allow type inference within templates?

**1. _SuccessType_ inference  (productStore.fetchProducts source Code command + Left Click )**
<img width="731" alt="image" src="https://user-images.githubusercontent.com/25116734/163086609-3c1de499-a989-451d-a770-ecf44184d7cf.png">

Full SourceCode : https://github.com/Shin-Beomchul/mono-vue3-module-federation
FileName : PiniaStoreExample.vue  

**2. _can not_ Type inference (productStore.fetchProducts source Code command + Left Click )**
<img width="721" alt="image" src="https://user-images.githubusercontent.com/25116734/163086641-0f45cd0e-bae6-42a6-88ea-af201fac54b0.png">
Full SourceCode : https://github.com/Shin-Beomchul/mono
FileName : Store.vue  

### Expected behavior

<img width="721" alt="image" src="https://user-images.githubusercontent.com/25116734/163086641-0f45cd0e-bae6-42a6-88ea-af201fac54b0.png">

<template>
...
productStore.fetchProducts()
...
</template>

 productStore.fetchProducts part command + Left Click then, I wish type inference was possible.
**like this**
<img width="731" alt="image" src="https://user-images.githubusercontent.com/25116734/163086609-3c1de499-a989-451d-a770-ecf44184d7cf.png">

### Actual behavior

can not_ Type inference of productStore.fetchProducts() ( command + Left Click)

**Comment by posva** [comment, marked as answer]
Have you tried using volar **instead** of vetur? in vscode

**Reply by Shin-Beomchul** [reply]
[posva](https://github.com/posva) Answer It's the perfect solution. Thank you
